### PR TITLE
$between: [Date1, Date2] does not work correcty in association's before hook

### PR DESCRIPTION
**Pull request by matthiasklan**
if changing parameters of the range, the new parameters won't get integrated into the SQL
(see issue [524#](https://github.com/mickhansen/graphql-sequelize/issues/524#issuecomment-334933013)). To run the tests use the test:between script

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
     "build:docker": "docker-compose build",
     "test:docker": "DIALECT=${DIALECT:=postgres} docker-compose run dev /bin/sh -c \"npm run test:integration\"",
     "test:integration": "mocha $npm_package_options_mocha test/integration/*.test.js test/integration/**/*.test.js",
+    "test:between": "mocha $npm_package_options_mocha test/integration/between.test.js",
```
Comment (mickhansen) [inline package.json]: not needed, mocha supports `.only` etc

**Comment by mickhansen** [comment]
@matthiasklan Please see if you can make the code much smaller, all the code can't possible be required to show the defect

**Comment by mickhansen** [comment]
```
{
  "createdAt": {
    "$between": [
      "2014-06-10T00:00:00.000Z",
      "2014-06-15T00:00:00.000Z"
    ]
  }
}
{
  "createdAt": {
    "$between": [
      "2014-06-10T00:00:00.000Z",
      "2014-06-18T00:00:00.000Z"
    ]
  }
}
```

Where statements look fine to me, your test is probably incorrect.

File: test/integration/between.test.js [hunk 2]
```diff
@@ -0,0 +1,424 @@
...
+                projectId: this.projectB.id
+              }
+            ]
+          }, {
+            include: [User.Tasks]
+          })
+        ).bind(this).spread(function (userA, userB) {
+          this.userA = userA;
+          this.userB = userB;
+          this.users = [userA, userB];
```
Comment (mickhansen) [inline test/integration/between.test.js]: Not using userB, why create it?

**Comment by mickhansen** [comment]
```
{
  "createdAt": {
    "$between": [
      "2014-06-10T00:00:00.000Z",
      "2014-06-18T00:00:00.000Z"
    ]
  }
}
Executing (default): SELECT `id`, `title`, `created_at` AS `createdAt`, `userId`, `projectId` FROM `tasks` AS `task` WHERE (`task`.`userId` IN (1) AND `task`.`created_at` BETWEEN '2014-06-10 00:00:00.000 +00:00' AND '2014-06-15 00:00:00.000 +00:00') ORDER BY `task`.`id` ASC;
```
SQL is incorrect though, odd.

File: test/integration/between.test.js (hunk 2 above)
Comment (matthiasklan) [reply test/integration/between.test.js]: I will remove all the unneeded code.

**Comment by mickhansen** [comment]
@matthiasklan Please cut the code way down, going through so many lines of irrelevant code is annoying and a waste of time.

**Comment by mickhansen** [comment]
The issue is resolved if direct input are used instead of graphql variables.

**Comment by matthiasklan** [comment]
Ok I will try to replace GraphQLDate with GraphQLString in my project. thank you very much for your help. The bug is really odd, because it works as expected, if I use the exact same before hook directly on Tasks filtered by createdAt and the userID.

**Comment by mickhansen** [comment]
The bug must be related to how caching is done in dataloader-sequelize and somehow date objects/GraphQLDate break it

**Comment by mickhansen** [comment]
Fixed in dataloader-sequelize@1.5.3

**Pull request closed without merging**